lib/vector-core/src/event/lua/metric.rs

```diff
@@ -82,33 +105,58 @@ impl<'a> FromLua<'a> for MetricTags {
     }
 }
 
-impl<'a> ToLua<'a> for MetricTags {
+impl<'a> ToLua<'a> for LuaMetricTags {
     fn to_lua(self, lua: &'a Lua) -> LuaResult<LuaValue> {
-        self.0.to_lua(lua)
+        if self.multi_value_tags {
+            Ok(LuaValue::Table(lua.create_table_from(
+                self.tags.0.into_iter().map(|(key, value)| {
+                    let value: Vec<_> = value
+                        .into_iter()
+                        .filter_map(|tag_value| tag_value.into_option().to_lua(lua).ok())
+                        .collect();
```

Comment:
You can expose errors from the inner `to_lua` through a special `collect`:
```suggestion
                    let value: Vec<_> = value
                        .into_iter()
                        .filter_map(|tag_value| tag_value.into_option().to_lua(lua))
                        .collect::<Result<Vec<_>, _>()?;
```

Reply:
Unfortunately the `?` can't be used since all of this code is inside of a `map` closure. I just stuck with the `filter_map` and `.ok()` since I believe the metric values will never fail to convert into Lua types.